roomserver/internal/input/input_events.go

```diff
@@ -128,9 +128,9 @@ func (r *Inputer) processRoomEvent(
 	if roomInfo == nil && !isCreateEvent {
 		return fmt.Errorf("room %s does not exist for event %s", event.RoomID(), event.EventID())
 	}
-	_, senderDomain, err := gomatrixserverlib.SplitID('@', event.Sender())
+	sender, err := event.UserID()
```

Comment:
This won't always work, but it will mostly work. We need to gracefully degrade when `err` is non nil.